ycmd/tests/get_completions_test.py

```diff
@@ -330,9 +1129,17 @@ def GetCompletions_CsCompleter_NonForcedReturnsNoResults_test():
                                   column_num = 21,
                                   force_semantic = False,
                                   query = 'Date' )
-  results = app.post_json( '/completions', completion_data ).json[ 'completions' ]
-
-  assert_that( results, empty() )
+  results = app.post_json( '/completions', completion_data ).json
+
+  # there are no semantic completions. However, we fall back to identifier
+  # completer in this case.
```

Comment:
You've been looking at these tests more recently than I have and you make good points; I'll leave the decision of what tests should be changed up to you. I trust your judgment.